packages/cli/cli/src/cli-context/upgrade-utils/getFernUpgradeMessage.ts

```diff
@@ -0,0 +1,162 @@
+            " \n(Run " +
+            chalk.cyan(`${cliEnvironment.cliName} upgrade`) +
+            " to update)";
+        upgradeAvailable = true;
+    }
+
+    // To start we're truncating the list and recommending the user use a different command
+    // to see the full list so as to not overwhelm them
+    message += getGeneratorUpgradeMessage({
+        generatorUpgradeInfo: upgradeInfo.generatorUpgradeInfo,
+        limit: 2,
+        includeBoxen: false
+    });
+    const generatorsNeedingUpgrades = upgradeInfo.generatorUpgradeInfo.filter((gui) => gui.isUpgradeAvailable);
+    if (generatorsNeedingUpgrades.length > 0) {
+        message += `\nRun ${chalk.cyan("fern generator upgrade")} to upgrade your generators.`;
+        upgradeAvailable = true;
+    }
+    if (generatorsNeedingUpgrades.length > 2) {
+        message +=
+            `\nRun ${chalk.cyan("fern generator upgrade --list")}` +
+            " to see the full list of generator upgrades available.";
+    }
+
+    return upgradeAvailable
+        ? boxen(message, {
+              padding: 1,
+              float: "center",
+              textAlignment: "center",
+              borderColor: "yellow",
+              borderStyle: "round"
+          })
+        : undefined;
+}
+
+export function getGeneratorUpgradeMessage({
+    generatorUpgradeInfo,
+    header,
+    limit,
+    includeBoxen = true
+}: {
+    generatorUpgradeInfo: FernGeneratorUpgradeInfo[];
+    header?: string;
+    limit?: number;
+    includeBoxen?: boolean;
+}): string | undefined {
+    let message = header ?? "";
+    let generatorUpgradeNeeded = false;
+
+    let generatorsNeedingUpgrades = generatorUpgradeInfo.filter((gui) => gui.isUpgradeAvailable);
+    if (limit != null) {
+        generatorsNeedingUpgrades = generatorsNeedingUpgrades.slice(0, limit + 1);
+    }
+    for (const generatorUpgrade of generatorsNeedingUpgrades) {
+        // Format => Generator 1.0.0 → 1.1.0 (API: API Name, Group: Group Name)
+        // ex: "Python SDK 1.0.0 → 1.1.0 (API: myApi, Group: myGroup)"
+        message +=
+            `\n${normalizeGeneratorName(generatorUpgrade.generatorName)} (${
+                generatorUpgrade.apiName != null ? "API: " + generatorUpgrade.apiName + ", " : ""
+            }Group: ${generatorUpgrade.generatorGroup}) ` +
+            chalk.dim(generatorUpgrade.currentVersion) +
+            chalk.reset(" → ") +
+            chalk.green(generatorUpgrade.latestVersion);
+
+        generatorUpgradeNeeded = true;
+    }
+
+    message += "\n";
+
+    if (generatorUpgradeNeeded) {
+        return includeBoxen
+            ? boxen(message, {
+                  padding: 1,
+                  float: "center",
+                  textAlignment: "center",
+                  borderColor: "yellow",
+                  borderStyle: "round"
+              })
+            : message;
+    }
+    return;
+}
+
+function normalizeGeneratorName(generatorName: string): string {
```

Comment:
Not actionable here, but it'd be great if we could define this generate this from somewhere as an enum so that we know exactly when we need to update it (e.g. when PHP is available, it'd be great if this was a compile error).

Reply:
I added an API to do this (you can update display name in the generators' seed.yml) so will update to that!